tests/testthat/test-multilevelannotationstep4.R

```diff
@@ -2,43 +2,65 @@ patrick::with_parameters_test_that(
   "multilevelannotation step 4 works",
   {
     # load data needed during step 4
-    load("../../data/adduct_weights.rda")
+    testdata_dir <- file.path(getwd(), "testdata", subfolder)
+    load(file.path(testdata_dir, "tempobjects.Rda"))
 
     testthat_wd <- getwd()
-    test_directory <- file.path(getwd(), "testdata/multilevelannotationstep4", subfolder)
+    outloc <- file.path(
+      tempdir(),
+      "multilevelannotationstep4",
+      subfolder
+    )
 
     # create test folder and copy necessary files
-    dir.create(test_directory, recursive = TRUE)
+    dir.create(outloc, recursive = TRUE)
     file.copy(
-      file.path(getwd(), "testdata/advanced", subfolder, "Stage3.csv"),
-      file.path(test_directory, "Stage3.csv")
+      file.path(testdata_dir, "Stage3.csv"),
+      file.path(outloc, "Stage3.csv")
     )
 
     # load expected results
-    expected <- read.csv(file.path(getwd(), "testdata/advanced", subfolder, "Stage4.csv"))
-    expected$MatchCategory <- as.character(expected$MatchCategory)
-
-    # set correct working directory
-    setwd(test_directory)
+    expected <- read.csv(file.path(testdata_dir, "Stage4.csv"))
+    #expected$MatchCategory <- as.character(expected$MatchCategory)
```

Comment:
This is not needed anymore? For some reason, the `MatchCategory` in read csv file had a different type. `dataCompareR::rCompare` can ignore this?

Reply:
I changed it to also read the actual results from file, so that the columns are actually of the same type.